sdk/core/azure-core-serializer-json-jackson/src/main/java/com/azure/core/serializer/json/jackson/JacksonJsonSerializer.java

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.serializer.json.jackson;
+
+import com.azure.core.serializer.JsonSerializer;
+import com.azure.core.util.logging.ClientLogger;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import reactor.core.Exceptions;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Objects;
+
+/**
+ * Jackson based implementation of the {@link JsonSerializer} interface.
+ */
+public final class JacksonJsonSerializer implements JsonSerializer {
+    private final ClientLogger logger = new ClientLogger(JacksonJsonSerializer.class);
+
+    private final ObjectMapper mapper;
+
+    /**
+     * Constructs a {@link JsonSerializer} using the passed Jackson serializer.
+     *
+     * @param mapper Configured Jackson serializer.
+     */
+    JacksonJsonSerializer(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    @Override
+    public <T> T read(byte[] input, Class<T> clazz) {
+        if (input == null) {
+            return null;
+        }
+
+        try {
+            return mapper.readValue(input, clazz);
```

Comment:
I don't remember of top of my head, does this support ObjectNode?

i.e., `mapper.readvalue(input, ObjectNode.class)`

Reply:
I believe that would be ObjectMapper.readTree, would this be a common scenario for Cosmos?

Reply:
yes, that's a common scenario.

Reply:
how about adding a if/else branch to the same method on the class type?

Reply:
Is the tree reading logic used when dealing with customer related objects or when dealing with service objects? I updated the PR description to better clarify the purpose of the interface added. If the tree reading scenario is common with customer object I'll dive into it more, I'd also rather see it being a separate API than tied into the `read` API which is more data-bindy.

Reply:
ObjectNode is used for both internal cosmos types, and also for user apis.

as cosmos is a schema less database, we also would like to provide a schema less type, for that reason we also support ObjectNode in the public surface area. The other reason for supporting ObjectNode in the public surface area is perf as deserialization to ObjectNode doesn't require reflection whereas for deserializing to user provivded Pojo, jackson will use reflection which is slow.

As an example all of the following public API invocation currently work:

```java
UserPojo userPojo = cosmosContainer.readItem(id, pk, UserPojo.class).block().getItem();
ObjectNode objectNode = cosmosContainer.readIteam(id, pk, ObjectNode.class).blcok.getItem();
```

Reply:
btw, at least for jackson, ObjectMapper is capable of handling ObjectNode under the same API:

```java
 static class UserPojo {
        @JsonProperty
        String id;

        @JsonProperty
        String pk;
    }

    public static void main(String[] args) throws Exception {

        UserPojo userPojo = new UserPojo();
        userPojo.id = "myId";
        userPojo.pk = "mypk";

        ObjectMapper objectMapper = new ObjectMapper();
        byte[] bytes = objectMapper.writeValueAsBytes(userPojo);

        ObjectNode deserializedObjectNode = objectMapper.readValue(bytes, ObjectNode.class);
        UserPojo deserializedPojo = objectMapper.readValue(bytes, UserPojo.class);

        assert (deserializedObjectNode.get("id").asText().equals(deserializedPojo.id));

        System.out.println("done");
    }
```

Reply:
@moderakh so support for serializing and deserializing `ObjectNode` is needed? I did some digging into the Jackson code and it does have a quick path if `ObjectNode.class` is passed as the `Class<T>` into the deserialize methods. I'll add additional tests to the PR which test this out.

Still digging into what GSON does, they do have a similar class `JsonObject` which I imagine also gets the same treatment.